WebKit/WebKit
# Move GCC_WARN_NON_VIRTUAL_DESTRUCTOR to CommonBase.xcconfig

#### 0b8e5ea6809c446d5793706d6958019e4f6716db
<pre>
Move GCC_WARN_NON_VIRTUAL_DESTRUCTOR to CommonBase.xcconfig
&lt;<a href="https://bugs.webkit.org/show_bug.cgi?id=285220">https://bugs.webkit.org/show_bug.cgi?id=285220</a>&gt;
&lt;<a href="https://rdar.apple.com/142127164">rdar://142127164</a>&gt;

Reviewed by Mike Wyrzykowski.

* Configurations/CommonBase.xcconfig:
(GCC_WARN_NON_VIRTUAL_DESTRUCTOR): Add.
* Source/JavaScriptCore/Configurations/Base.xcconfig:
(GCC_WARN_NON_VIRTUAL_DESTRUCTOR): Delete.
* Source/ThirdParty/gmock/Configurations/Base.xcconfig:
(GCC_WARN_NON_VIRTUAL_DESTRUCTOR): Delete.
* Source/WTF/Configurations/Base.xcconfig:
(GCC_WARN_NON_VIRTUAL_DESTRUCTOR): Delete.
* Source/WebCore/Configurations/Base.xcconfig:
(GCC_WARN_NON_VIRTUAL_DESTRUCTOR): Delete.
* Source/WebCore/PAL/Configurations/Base.xcconfig:
(GCC_WARN_NON_VIRTUAL_DESTRUCTOR): Delete.
* Source/WebCore/PAL/ThirdParty/dav1d/Configurations/Base.xcconfig:
(GCC_WARN_NON_VIRTUAL_DESTRUCTOR): Delete.
* Source/WebGPU/Configurations/Base.xcconfig:
(GCC_WARN_NON_VIRTUAL_DESTRUCTOR): Delete.
* Source/WebInspectorUI/Configurations/Base.xcconfig:
(GCC_WARN_NON_VIRTUAL_DESTRUCTOR): Delete.
* Source/WebKit/Configurations/Base.xcconfig:
(GCC_WARN_NON_VIRTUAL_DESTRUCTOR): Delete.
* Source/WebKitLegacy/mac/Configurations/Base.xcconfig:
(GCC_WARN_NON_VIRTUAL_DESTRUCTOR): Delete.
* Source/bmalloc/Configurations/Base.xcconfig:
(GCC_WARN_NON_VIRTUAL_DESTRUCTOR): Delete.

* Tools/DumpRenderTree/mac/TestRunnerMac.mm:
(TestRunner::~TestRunner):
- Use default.
* Tools/TestRunnerShared/UIScriptContext/UIScriptContext.h:
(WTR::UIScriptContextDelegate::UIScriptContextDelegate): Add.
(WTR::UIScriptContextDelegate::~UIScriptContextDelegate): Add.
- Define default constructor and virtual destructor.
* Tools/TestWebKitAPI/Tests/WTF/WeakPtr.cpp:
(TestWebKitAPI::MultipleInheritanceBase1::MultipleInheritanceBase1): Add.
(TestWebKitAPI::MultipleInheritanceBase1::~MultipleInheritanceBase1): Add.
(TestWebKitAPI::MultipleInheritanceBase2::MultipleInheritanceBase2): Add.
(TestWebKitAPI::MultipleInheritanceBase2::~MultipleInheritanceBase2): Add.
(TestWebKitAPI::MultipleInheritanceDerived::MultipleInheritanceDerived): Add.
(TestWebKitAPI::MultipleInheritanceDerived::~MultipleInheritanceDerived): Add.
- Define default constructor and virtual destructor.
(TestWebKitAPI::Dog::~Dog):
- Make destructor virtual.
* Tools/WebGPUPlayground/Configurations/Base.xcconfig:
(GCC_WARN_NON_VIRTUAL_DESTRUCTOR): Delete.
* Tools/WebKitTestRunner/InjectedBundle/TestRunner.cpp:
(WTR::TestRunner::~TestRunner):
- Use default.
* Tools/lldb/lldbWebKitTester/Configurations/Base.xcconfig:
(GCC_WARN_NON_VIRTUAL_DESTRUCTOR): Delete.

Canonical link: <a href="https://commits.webkit.org/288332@main">https://commits.webkit.org/288332@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/85f505af71417d669d73210986ccd295d9bf9b3c

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/82472 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/2134 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/36564 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/87603 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/33533 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/84575 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/2205 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/10027 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/64277 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/22032 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/85539 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/1601 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/75031 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/44554 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/1471 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/29218 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/32572 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/75460 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/72763 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/29855 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/88961 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/81526 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/9776 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/7035 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/72679 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/10002 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/70845 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/71896 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/16036 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/15044 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/1158 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/12808 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/9729 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/15250 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/103937 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/9603 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/25214 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/13069 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/11373 "Built successfully") | | |
